xrpl/models/requests/tx.py

```diff
@@ -17,18 +17,28 @@
 class Tx(Request):
     """
     The tx method retrieves information on a single transaction.
+    The Request must contain either transaction or CTID parameter, but not both.
 
     `See tx <https://xrpl.org/tx.html>`_
     """
 
     method: RequestMethod = field(default=RequestMethod.TX, init=False)
-    transaction: str = REQUIRED  # type: ignore
-    """
-    This field is required.
-
-    :meta hide-value:
-    """
-
+    transaction: Optional[str] = None
     binary: bool = False
     min_ledger: Optional[int] = None
     max_ledger: Optional[int] = None
+    ctid: Optional[str] = None
+
+    def _get_errors(self: Tx) -> Dict[str, str]:
+        errors = super()._get_errors()
+        if not self._has_only_one_input():
+            errors[
+                "Tx"
+            ] = "Must have only one of `ctid` or `transaction`, but not both."
+        return errors
+
+    def _has_only_one_input(self: Tx) -> bool:
+        present_items = [
+            item for item in [self.transaction, self.ctid] if item is not None
+        ]
+        return len(present_items) == 1
```

Comment:
Instead of list comprehension here, you can use `filter` to remove items which don't pass a test. If you're just trying to remove falsey values like `None` `bool` is a good enough function. (https://www.geeksforgeeks.org/remove-falsy-values-from-a-list-in-python/)

Explanation of truthy / falsey values in python: [truthy](https://www.freecodecamp.org/news/truthy-and-falsy-values-in-python/)

And last note, for readability I'd just recommend pulling the array `[self.transaction, self.ctid]` into a separate variable as there's already a lot going on in the one-liner, and that'll allow you to label why those two variables are being grouped. Ex. `unique_ids = [self.transaction, self.ctid]`

Reply:
okay, fixed in [ce890c0](https://github.com/XRPLF/xrpl-py/pull/672/commits/ce890c04ceb5fc35ef832fc1c8439e712c4cd3e0)